File: runtime/ruby/units/table/call.rb
# requires: rt/trap
# `type_key` is a symbol interned from the function type's shape (not a
# type index): tables can be shared across modules, whose index spaces
# differ.
def call(i, type_key, *args)
  Rt.trap("undefined element") if i >= @slots.size
  slot = @slots[i]
  Rt.trap("uninitialized element") if slot.nil?
  ty, func = slot
  Rt.trap("indirect call type mismatch") unless ty == type_key
  func.call(*args)
end

File: runtime/python/units/table/call.py
# requires: rt/trap
def call(self, i, type_key, *args):
    if i >= len(self._slots):
        Rt.trap("undefined element")
    slot = self._slots[i]
    if slot is None:
        Rt.trap("uninitialized element")
    ty, func = slot
    if ty != type_key:
        Rt.trap("indirect call type mismatch")
    return func(*args)
